app/models/concerns/system_like.rb

```diff
@@ -9,7 +9,9 @@ module SystemLike
     scoped_search on: %i[id name]
     has_many :rule_results, dependent: :destroy
     has_many :rules, through: :rule_results, source: :rule
-    belongs_to :account, optional: true
+    belongs_to :account_object, optional: true, foreign_key: :account,
+                                primary_key: :account_number,
+                                inverse_of: :hosts, class_name: 'Account'
```

Comment:
The column name in the cyndi view is `account`, so we can't have a method/relationship named the same, unfortunately.

Reply:
We cannot change the DB view provided to us by Cyndi in any way, even column aliases. One option would be to create a `hosts` view within our own schema that points to the cyndi hosts view and renames fields. A view of a view... seems not so great to me, but it's a possibility. IMO it's not a huge problem within our code to account for this (pun intended :) )

Reply:
In fact, it makes way more sense to use `account_number` instead of `id`, and ideally, we will be able to completely remove the accounts table if, for every related record, we store the `account_number` directly.